packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -156,6 +155,13 @@ class AOTSnapshotter {
       genSnapshotArgs.add('--no-use-integer-division');
     }
 
+    // Optimization arguments.
+    genSnapshotArgs.addAll(<String>[
+      // Faster async/await
+      '--no-causal-async-stacks',
```

Comment:
We should also update the flags to gen_snapshot in `fuchsia/fuchsia_build.dart` and `dart_component.gni` in the Fuchsia tree to ensure we're doing the same thing everywhere. @iskakaushik